submissions/DaniaB24/frogger-game/app.js

```diff
@@ -0,0 +1,125 @@
+// Enemies our player must avoid
+const FIELD_WIDTH = 505,
+  START_X = 202,
+  START_Y = 404;
+CELL_WIDTH = 101;
+CELL_HEIGHT = 83;
+let count = 0;
+var Enemy = function (x, y, speed, player) {
+  // Variables applied to each of our instances go here,
+  // we've provided one for you to get started
+  this.x = x;
+  this.y = y;
+  this.speed = speed;
+  this.player = player;
+  // The image/sprite for our enemies, this uses
+  // a helper we've provided to easily load images
+  this.sprite = "images/enemy-bug.png";
+};
+
+const Player = function (x, y) {
+  this.x = x;
+  this.y = y;
+  this.sprite = "images/char-boy.png";
+};
+// Update the enemy's position, required method for game
+// Parameter: dt, a time delta between ticks
+Enemy.prototype.update = function (dt) {
+  // You should multiply any movement by the dt parameter
+  // which will ensure the game runs at the same speed for
+  // all computers.
+  if (this.x < FIELD_WIDTH) {
+    this.x += this.speed * dt;
+  } else {
+    this.x += -FIELD_WIDTH;
+  }
+  if (this.checkCollision()) player.resetPosition();
+};
+Enemy.prototype.checkCollision = function () {
+  return (
+    this.y + CELL_HEIGHT > this.player.y &&
+    this.player.x < this.x + CELL_WIDTH &&
+    this.player.x > this.x - CELL_WIDTH
+  );
+};
+// Draw the enemy on the screen, required method for game
+Enemy.prototype.render = function () {
+  ctx.drawImage(Resources.get(this.sprite), this.x, this.y);
+};
+
+// Now write your own player class
+// This class requires an update(), render() and
+// a handleInput() method.
+Player.prototype.update = function () {
+  document.getElementById('score').innerHTML = 
+  `
+    <div>
+      <p>Score:<span>${count}</span></p>
+    </div>
+  `
```

Comment:
The class shouldn't be aware of the specificity of the environment it operates in.

The game engine doesn't offer an appropriate method to us. Let's define a global function for this and pretend it is a part of the game engine.

Reply:
thanks for review! I already fix)